src/display/canvas.js

```diff
@@ -477,7 +473,8 @@ var CanvasGraphics = (function CanvasGraphicsClosure() {
     this.smaskStack = [];
     this.smaskCounter = 0;
     this.tempSMask = null;
-    this.cachedCanvases = new CachedCanvases();
+    this.canvasFactory = canvasFactory || new DOMCanvasFactory();
```

Comment:
`canvasFactory` parameter must be always specified (starting at render() method). can you also move `canvasFactory` before `imageLayer` in the parameters list.